tests/sles4sap/bone_install.pm

```diff
@@ -0,0 +1,76 @@
+# SUSE's SLES4SAP openQA tests
+#
+# Copyright 2019-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Install SAP Business One via command line.
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+use base 'sles4sap';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use Utils::Backends;
+use utils qw(file_content_replace zypper_call);
+use Utils::Systemd 'systemctl';
+use version_utils 'is_sle';
+use POSIX 'ceil';
+use Utils::Logging 'save_and_upload_log';
+
+sub run {
+    my ($self) = @_;
+    my ($proto, $path) = $self->fix_path(get_required_var('BONE'));
+    my $sid = get_required_var('INSTANCE_SID');
+    my $instid = get_required_var('INSTANCE_ID');
+    # set timeout as 4800 as a temp workaround for slow nfs
+    my $tout = get_var('HANA_INSTALLATION_TIMEOUT', 4800);    # Timeout for HANA installation commands.
+
+    select_serial_terminal;
+    #select_console 'x11';
+
+    # Transfer media.
+    my $target = '/sapinst';    # Directory in SUT where install media will be copied token
+    if (get_var 'ASSET_0') {
```

Comment:
Schedule calls both `hana_install` and `bone_install`. While going the NFS route, `hana_install` downloads media from whatever the *HANA* setting points to and then `bone_install` searches for it in *BONE*; however in the *ASSET_0* route, both modules are  in the *ASSET_0* route, both modules are downloading the installer from the same setting. So, questions:

* Is the intention to store in osd an asset/tarball that contains the installer for HANA and Business One together?
* Won't these calls to `download_hana_assets_from_server* in `hana_install` and then in `bone_install` mean that the job will download the same tarball twice? If so, perhaps `bone_install` should not download it, but only check that it's there.

Reply:
My attempt to fix by using: unless (-e $asset_lock) {
        assert_script_run "wget -O - $hana_location | tar -xf -", timeout => $nettout;
        script_run "touch $asset_lock";
}
Didn't work. On a local perl script, it checks if the file exists, but on osd, it seems to just check if the variable exists. I don't know what I'm doing wrong here.

Reply:
@alvarocarvajald suggestion fixed the issue.